pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -148,6 +149,63 @@ func (r *hibernationReconciler) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, nil
 	}
 
+	// Check for the hibernate-after annotation, and if the cluster has been in running state for longer than this duration, put it to sleep.
+	hibernateAfter, ok := cd.Annotations[constants.HibernateAfterAnnotation]
+	if ok && cd.Spec.PowerState != hivev1.HibernatingClusterPowerState {
+		cdLog.Debugf("found hibernate-after annotation: %s", hibernateAfter)
+		dur, err := time.ParseDuration(hibernateAfter)
```

Comment:
Use a more descriptive variable name for `dur`, since it is used later when the reader may have forgotten what the variable is.